./configure doesn't check for pandoc

HI,

./configure completes succesfully without pandoc, but building doesn't:

```
  CXXLD    libpipebackend.la
make[3]: Leaving directory `/data/home/cyclops/dev/pdns/modules/pipebackend'
make[3]: Entering directory `/data/home/cyclops/dev/pdns/modules'
make[3]: Nothing to be done for `all-am'.
make[3]: Leaving directory `/data/home/cyclops/dev/pdns/modules'
make[2]: Leaving directory `/data/home/cyclops/dev/pdns/modules'
Making all in codedocs
make[2]: Entering directory `/data/home/cyclops/dev/pdns/codedocs'
make[2]: Nothing to be done for `all'.
make[2]: Leaving directory `/data/home/cyclops/dev/pdns/codedocs'
Making all in docs
make[2]: Entering directory `/data/home/cyclops/dev/pdns/docs'
echo "You need pandoc to generate the manpages"
You need pandoc to generate the manpages
exit 1
make[2]: *** [pdnsutil.1] Error 1
make[2]: Leaving directory `/data/home/cyclops/dev/pdns/docs'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/data/home/cyclops/dev/pdns'
make: *** [all] Error 2
```

--- a/docs/Makefile.am
+++ b/docs/Makefile.am
@@ -24,38 +24,41 @@ MANPAGES_TARGET_TOOLS += dnsbulktest.1 \
 	dnstcpbench.1
 endif
 
-
 if HAVE_PROTOBUF
 if HAVE_PROTOC
 MANPAGES_TARGET_TOOLS += dnspcap2protobuf.1
 endif
 endif
 
-MANPAGES_TARGET_DNSDIST = dnsdist.1
+EXTRA_DIST = manpages markdown/*.md markdown/appendix markdown/authoritative markdown/common markdown/httpapi markdown/recursor markdown/security markdown/tools $(MANPAGES_TARGET_AUTH) $(MANPAGES_TARGET_TOOLS)
 
-MANPAGES_TARGET_RECURSOR = pdns_recursor.1 \
-	rec_control.1
+# Figure out the manpages to build/install
+MANPAGES_TARGET_ALL=$(MANPAGES_TARGET_AUTH)
+man_MANS = $(MANPAGES_TARGET_AUTH)
 
-MANPAGES_TARGET_ALL=$(MANPAGES_TARGET_AUTH) \
-		$(MANPAGES_TARGET_RECURSOR) \
-		$(MANPAGES_TARGET_TOOLS) \
-		$(MANPAGES_TARGET_DNSDIST)
+if TOOLS
+MANPAGES_TARGET_ALL += $(MANPAGES_TARGET_TOOLS)
+man_MANS += $(MANPAGES_TARGET_TOOLS)
+endif
 
-# The manpages to distribute and install are only those for the
-# auth server and tools. For the recursor and dnsdist tarballs,
-# the respective dist- scripts will take care of this
+all-manpages: $(MANPAGES_TARGET_ALL)
+
+# If we can't rebuild the manpages, don't allow creation or deletion
 if HAVE_PANDOC
-dist_man_MANS = $(MANPAGES_TARGET_AUTH) $(MANPAGES_TARGET_TOOLS)
+$(MANPAGES_TARGET_ALL): %: manpages/%.md
+	$(PANDOC) -s -t man $< -o $@
+
+clean:
+	rm -rf html html.tar.bz2 *.8 *.1
+else
+if !HAVE_MANPAGES
+$(MANPAGES_TARGET_ALL):
+	echo "You need pandoc to generate the manpages"
+	exit 1
 endif
-if HAVE_MANPAGES
-dist_man_MANS = $(MANPAGES_TARGET_AUTH) $(MANPAGES_TARGET_TOOLS)
 endif
 
-EXTRA_DIST = manpages markdown/*.md markdown/appendix markdown/authoritative markdown/common markdown/httpapi markdown/recursor markdown/security markdown/tools
-
-
-.PHONY: html all-manpages
-
+# HTML documentation
 html: html/index.html
 
 if FROM_GIT
@@ -82,16 +85,4 @@ html/index.html:
 	@echo "supported from a git checkout"
 endif
 
-all-manpages: $(MANPAGES_TARGET_ALL)
-
-if HAVE_PANDOC
-$(MANPAGES_TARGET_ALL): %: manpages/%.md
-	$(PANDOC) -s -t man $< -o $@
-else
-$(MANPAGES_TARGET_ALL):
-	echo "You need pandoc to generate the manpages"
-	exit 1
-endif
-
-clean:
-	rm -rf html html.tar.bz2 *.8 *.1
+.PHONY: html all-manpages

